### PR TITLE
Story/vspc 85

### PR DESCRIPTION
**Pull request by Sagar-k93**
On the staff space page, there should be a list of linked spaces. The list should show all the spaces that are linked via SpaceLinks, including both the links on the space and links from other spaces to this space.

### CODE REVIEW

File: vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java [hunk 1]
```diff
@@ -217,11 +217,11 @@ public ISpace getFullyLoadedSpace(String id) {
     public void deleteSpaceById(String id) {
         if(id != null) {
             List<SpaceLink> spaceLinks = spaceLinkRepo.getLinkedSpaces(id);
-            List<SpaceLink> linksFromOtherSpaces = spaceLinkRepo.getLinkedFromSpaces(id);
+            List<SpaceLink> fromSpaceLinks = spaceLinkRepo.getLinkedFromSpaces(id);
             Exhibition exhibition = (Exhibition) exhibitionManager.getStartExhibition();
             // When space has other links attached to it
             // To delete links that access to the space getting deleted and replacing it as null
-            for(SpaceLink sp : linksFromOtherSpaces) {
+            for(SpaceLink sp : fromSpaceLinks) {
```
Comment (prjadhav14) [inline vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java:224]: we can say variable as space may be?

File: vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.jsp [hunk 2]
```diff
@@ -1100,10 +1098,31 @@ ${space.description}
...
+</ul>
+</c:if>
+<c:if test="${not empty linksToThisSpace}">
+<h5 style ="overflow: hidden; padding-left: 0.5em">Space links to this space:</h5>
+<ul style ="overflow: hidden;">
+<c:forEach items="${linksToThisSpace}" var="spaceLinks">
+<li><h6>${spaceLinks.sourceSpace.name}</h6></li>
+</c:forEach>
+</ul>
+</c:if>
```
Comment (prjadhav14) [inline vspace/src/main/webapp/WEB-INF/views/staff/spaces/space.jsp:1128]: indentation

File: vspace/src/main/java/edu/asu/diging/vspace/web/staff/SpaceController.java [hunk 3]
```diff
@@ -38,6 +38,8 @@ public String showSpace(@PathVariable String id, Model model) {
     	
     	
         ISpace space = spaceManager.getFullyLoadedSpace(id);
+        model.addAttribute("linksOnThisSpace", spaceManager.getLinkedSpaces(id));
+        model.addAttribute("linksToThisSpace",spaceManager.getLinkedFromSpaces(id));
```
Comment (jdamerow) [inline vspace/src/main/java/edu/asu/diging/vspace/web/staff/SpaceController.java:42]: method names are confusing, why not `getLinksToSpaces` and `getLinksFromSpaces`? Or even better 'getOutgoingLinks` and 'getIncomingLinks`?

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Comment by jdamerow** [comment]
- let's have the list of links always on the right side of the window (right aligned)
- and let's still have the headings if there are no links but just say "There are no links to this space." (or from this space)

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Comment by jdamerow** [comment]
a few more things:
- let's use the bootstrap list for the links (https://getbootstrap.com/docs/3.4/components/#list-group-buttons) (but maybe make the list a little smaller)
- and make the links go to the other linked space

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Comment by jdamerow** [comment]
sorry @Sagar-k93 I broke it with my adhoc changes. would you please merge and resolve conflicts.

**Comment by jdamerow** [comment]
Let's choose a color that doesn't hurt your eyes when you look at it (one that fits with the rest of the color scheme) ;)

also, the link list should always have the same width

image

**Comment by jdamerow** [comment]
Jenkins, test this please.

**Pull request merged**